message_generation/src/test/java/org/ros/internal/message/RawMessageSerializationTest.java

```diff
@@ -266,4 +267,104 @@ public void testFloat64Array() {
     rawMessage.setFloat64Array("data", new double[] { 1, 2, 3, 4, 5 });
     checkSerializeAndDeserialize(rawMessage);
   }
+  
+  @Test
+  public void testChannelBufferFixedSizeWithInitialization() {
+    topicDefinitionResourceProvider.add("foo/foo", "uint8[5] data");
+    ChannelBuffer buffer = MessageBuffers.dynamicBuffer();
+    buffer.writeBytes(new byte[] { 1, 2, 3, 4, 5 });
+    RawMessage rawMessage = messageFactory.newFromType("foo/foo");
+    rawMessage.setChannelBuffer("data", buffer);
+    checkSerializeAndDeserialize(rawMessage);
+  }
+
+  @Test
+  public void testChannelBufferFixedSizeWithIncompleteInitialization() {
+    topicDefinitionResourceProvider.add("foo/foo", "uint8[5] data");
+    ChannelBuffer buffer = MessageBuffers.dynamicBuffer();
+    buffer.writeBytes(new byte[] { 1, 2, 3 });
+    RawMessage rawMessage = messageFactory.newFromType("foo/foo");
+    rawMessage.setChannelBuffer("data", buffer);
+    checkSerializeAndDeserialize(rawMessage);
+  }
+
+  @Test
+  public void testChannelBufferFixedSizeNoInitialization() {
```

Comment:
What's the difference between this test and a simple byte array? Should we add a byte array test too?

Reply:
Found this in the `RawMessage` interface:
```
  /**
   * @deprecated replaced by {@link #setInt8(String, byte)}
   */
  void setByte(String name, byte value);

  /**
   * @deprecated replaced by {@link #setInt8Array(String, byte[])}
   */
  void setByteArray(String name, byte[] value);

  /**
   * @deprecated replaced by {@link #setUInt8(String, byte)}
   */
  void setChar(String name, short value);

  /**
   * @deprecated replaced by {@link #setUInt8Array(String, byte[])}
   */
  void setCharArray(String name, short[] value);
```

Reply:
Added the tests for byte arrays and they are failing too.

![image](https://user-images.githubusercontent.com/9289769/38758180-23897ad2-3f46-11e8-9f54-794963db4ba5.png)